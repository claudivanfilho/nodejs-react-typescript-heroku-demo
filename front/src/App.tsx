import { Route, Routes } from "react-router-dom";

import { IntlProviderLocal } from "./context/IntlContextLocal";
import { LocaleProvider } from "./context/LocaleContext";
import BaseLayout from "./layouts/BaseLayout";

export default function App() {
  return (
    <LocaleProvider>
      <IntlProviderLocal>
        <Routes>
          <Route path="/" element={<BaseLayout />} />
        </Routes>
      </IntlProviderLocal>
    </LocaleProvider>
  );
}
