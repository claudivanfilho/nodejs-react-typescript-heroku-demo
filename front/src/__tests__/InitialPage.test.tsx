import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import App from "../App";

describe("Test Initial Page", () => {
  it("should match with snapshot", () => {
    expect(
      render(
        <Router>
          <App />
        </Router>
      ).container
    ).toMatchSnapshot();
  });
});
