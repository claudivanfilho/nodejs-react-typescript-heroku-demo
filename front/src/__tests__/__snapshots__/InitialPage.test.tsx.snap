// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Initial Page should match with snapshot 1`] = `
<div>
  <div
    class="h-screen pt-14"
  >
    <div
      class="fixed top-0 z-10 w-full py-3 text-white bg-green-700"
    >
      <div
        class="flex justify-between max-w-screen-xl px-5 mx-auto 2xl:px-0"
      >
        <h1
          class="text-2xl font-bold"
        >
          <span>
            PhoneWords
          </span>
        </h1>
        <select
          class="px-2 bg-transparent"
        >
          <option
            value="en"
          >
            EN
          </option>
          <option
            value="es"
          >
            ES
          </option>
          <option
            value="pt"
          >
            PT
          </option>
          <option
            value="ko"
          >
            KO
          </option>
          <option
            value="ja"
          >
            JA
          </option>
          <option
            value="fr"
          >
            FR
          </option>
          <option
            value="de"
          >
            DE
          </option>
        </select>
        <div
          class="flex items-center text-xl font-bold cursor-pointer lg:hidden"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          Back
        </div>
      </div>
    </div>
    <main
      class="overflow-y-auto"
      style="height: calc(100vh - 135px);"
    >
      Test
    </main>
    <footer
      class="fixed bottom-0 w-full py-2 text-sm font-bold text-center text-white bg-green-700"
    >
      Created by Claudivan
    </footer>
  </div>
</div>
`;
